game/groundforces/ground_force_group.py

```
@@ -0,0 +1,163 @@
+        except StopIteration:
+            logging.warning(f"Template {template.name} not usable by {faction.name}")
+
+    def load_templates(self, faction: Faction) -> None:
+        """Load all possible templates for this group"""
+        for template_name in self.template_names:
+            for template in db.TEMPLATES.by_name(template_name):
+                self.add_template(template, faction)
+
+        if not self._templates:
+            # Find all matching templates if no preferred set or available
+            for template in list(db.TEMPLATES.by_tasks(self.tasks)):
+                if any(self.has_unit_type(unit) for unit in template.units):
+                    self.add_template(template, faction)
+
+    def set_templates(self, templates: list[GroundObjectTemplate]) -> None:
+        self._templates = templates
+
+    def has_unit_type(self, unit_type: UnitType[Any]) -> bool:
+        return unit_type in self.ground_units or unit_type in self.ship_units
+
+    @property
+    def unit_types(self) -> Iterator[str]:
+        for unit in self.ground_units:
+            yield unit.dcs_id
+        for ship in self.ship_units:
+            yield ship.dcs_id
+        for static in self.statics:
+            yield static
+
+    @classmethod
+    def named(cls, name: str) -> GroundForceGroup:
+        if not cls._loaded:
+            cls._load_all()
+        return cls._by_name[name]
+
+    def generate(
+        self,
+        name: str,
+        position: PointWithHeading,
+        control_point: ControlPoint,
+        game: Game,
+        template_name: str = "",
+    ) -> Optional[TheaterGroundObject]:
+        """Try to generate a TheaterGroundObject from the group"""
+        if self.templates:
+            # Choose forced template or random one if no available or defined
+            template = self._template_by_name(template_name) or random.choice(
+                self.templates
+            )
```

This behavior matches the name of the argument here, but elsewhere it's named `forced_template`. If the behavior when that template isn't available is to fall back to a random choice, I think `preferred_template` is a better name.